apps/gdaltindex_lib.cpp

```diff
@@ -165,6 +166,231 @@ static bool PatternMatch(const char *input, const char *pattern)
+                "the target SRS defined by -t_srs."));
+
+    argParser->add_argument("-max_pixel_size")
+        .metavar("<val>")
+        .store_into(psOptions->dfMaxPixelSize)
+        .help(_("Maximum pixel size in term of geospatial extent per pixel "
+                "(resolution) that a raster should have to be selected. The "
+                "pixel size is evaluated after reprojection of its extent to "
+                "the target SRS defined by -t_srs."));
+
+    argParser->add_output_format_argument(psOptions->osFormat);
+
+    argParser->add_argument("-tileindex")
+        .metavar("<field_name>")
+        .store_into(psOptions->osLocationField)
+        .help(_("Name of the layer in the tile index file."));
+
+    argParser->add_argument("-write_absolute_path")
+        .flag()
+        .store_into(psOptions->bWriteAbsolutePath)
+        .help(_("Write the absolute path of the raster files in the tile index "
+                "file."));
+
+    argParser->add_argument("-skip_different_projection")
+        .flag()
+        .store_into(psOptions->bSkipDifferentProjection)
+        .help(_(
+            "Only files with the same projection as files already inserted in "
+            "the tile index will be inserted (unless -t_srs is specified)."));
+
+    argParser->add_argument("-t_srs")
+        .metavar("<target_srs>")
+        .store_into(psOptions->osTargetSRS)
+        .help(_("Geometries of input files will be transformed to the desired "
+                "target coordinate reference system."));
+
+    argParser->add_argument("-src_srs_name")
+        .metavar("<field_name>")
+        .store_into(psOptions->osSrcSRSFieldName)
+        .help(_("Name of the field in the tile index file where the source SRS "
+                "will be stored."));
+
+    argParser->add_argument("-src_srs_format")
+        .metavar("{AUTO|WKT|EPSG|PROJ}")
+        .choices("AUTO", "WKT", "EPSG", "PROJ")
+        .action(
+            [psOptions](const auto &f)
+            {
+                if (f == "WKT")
+                    psOptions->eSrcSRSFormat = FORMAT_WKT;
+                else if (f == "EPSG")
+                    psOptions->eSrcSRSFormat = FORMAT_EPSG;
+                else if (f == "PROJ")
+                    psOptions->eSrcSRSFormat = FORMAT_PROJ;
+                else
+                    psOptions->eSrcSRSFormat = FORMAT_AUTO;
+            })
+        .help(_("Format of the source SRS to store in the tile index file."));
+
+    argParser->add_argument("-lyr_name")
+        .metavar("<name>")
+        .store_into(psOptions->osIndexLayerName)
+        .help(_("Name of the layer in the tile index file."));
+
+    argParser->add_argument("-lco")
+        .metavar("<KEY>=<VALUE>")
+        .append()
+        .action(
+            [psOptions](const std::string &s)
+            {
+                if (psOptions)
+                    psOptions->aosLCO.AddString(s.c_str());
```

Comment:
psOptions can't be null:
```suggestion
                psOptions->aosLCO.AddString(s.c_str());
```